src/core/model_config.proto

```diff
@@ -623,6 +623,75 @@ message ModelOptimizationPolicy
   PinnedMemoryBuffer output_pinned_memory = 6;
 }
 
+//@@
+//@@.. cpp:var:: message RequestTimeoutPolicy
+//@@
+//@@   Timeout policy for inference requests. If specified, the scheduler will
+//@@   handle pending requests that exceed timeout accordingly.
+//@@
+message RequestTimeoutPolicy
+{
+  //@@
+  //@@  .. cpp:enum:: Action
+  //@@
+  //@@     The action applied on timeout request.
+  //@@
+  enum Action {
+    //@@    .. cpp:enumerator:: Action::REJECT = 0
+    //@@
+    //@@       Reject the request and return error message accordingly.
+    //@@
+    REJECT = 0;
+
+    //@@    .. cpp:enumerator:: Action::DELAY = 1
+    //@@
+    //@@       Delay the request until all other requests within the timeout
```

Comment:
"Delay the request until all other requests at the same (or higher) priority levels that have not reached their timeouts are processed. A delayed request will eventually be processed, but may be delayed indefinitely due to newly arriving requests."

Remove "The priority...". We need to allow a timeout policy for each priority level.